### PR TITLE
initial pass at cookbook

### PR DESCRIPTION
**Pull request by willejs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,24 @@ The cookbook includes a thorfile to encrypt your secrets into .crypt files, and
...
 
+# The idea
+
+Managing secrets is hard. Chef encrypted data bags are not ideal, and are reasonably hard to manage. Chef vault has overheads... So this is where KMS steps in.
+
+KMS enables you to create a private key in amazon, and control access to it. You can then call the API to encrypt and decrypt data, if you have access privileges to that key.
+
+Therefore, you can:
+  - Encrypt all your data into files using a key
+  - Bake the files into your amazon AMIs (or drop them off) and make sure they are owned by root...
```
Comment (lmunro) [inline README.md]: by drop them off do you mean pull them from a remote source like s3? Some clarification would be nice here.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+LineLength:
+  Enabled: false
```
Comment (lmunro) [inline .rubocop.yml]: line length issues are usually easy to fix, or just extend the length a little.  personal preference though so up to you.

**Comment by lmunro** [comment]
LGTM. One thing I would say is it would be nice to be able to choose to only decrypt the secrets that you need, so you don;t have a bunch of clear text secrets available on the server if it were to be compromised.  Otherwise nice work!

File: .rubocop.yml (hunk 2 above)
Comment (willejs) [inline .rubocop.yml]: 👍

**Comment by willejs** [comment]
@lmunro Cheers! Yeah, your right, it would be nice to specify files individually to decrypt, rather than all of them in future.
My idea was just to make it super simple to start, and later on be able to specify a directory, or list of files...
For some reason when i tried to unit test the methods in the helpers, they were not found from the class that i had included, so il fix that at some point and write some unit tests for the helpers and stub the kms methods out.

**Pull request merged**